macq/extract/esam.py

```diff
@@ -44,13 +49,21 @@ def __new__(cls,
                 obs_trace_list: ObservedTraceList = None,
                 debug=False,
                 recursion_limit=2000,
+                obj_to_sort: dict[str, Sort] = None,
+                sorts: list[Sort] = None,
+                action_2_sort: dict[str, list[str]] = None,
+                fluent_types: [str, list]=None,
                 **kwargs
                 ) -> Model:
         """ learns from fully observable observations under no further assumptions to extract a safe lifted action model
         of the problem's domain.
             Args:
                 obs_trace_list(ObservedTraceList): tokenized TraceList.
+                recursion_limit(int): maximum recursion limit, option to change recursion limit, essential when trying
```

Comment:
Why do we need this parameter?
Type annotations are done using typing class, read about it and the types it contains.